modules/collabora_online_group/tests/src/Functional/ViewTest.php

```diff
@@ -0,0 +1,143 @@
+      'group_type' => 'group_type_1',
+      'scope' => PermissionScopeInterface::INSIDER_ID,
+      'global_role' => RoleInterface::AUTHENTICATED_ID,
+      'permissions' => [
+        'view group',
+        'access group_media overview',
+        'view group_media:document entity',
+        'edit any group_media:document in collabora',
+        'preview group_media:document in collabora',
+        ],
+    ]);
+    $this->createPluginRelation($group_type, 'group_media:document', [
+      'group_cardinality' => 0,
+      'entity_cardinality' => 1,
+      'use_creation_wizard' => FALSE,
+    ]);
+
+    // Create content.
+    $group = $this->createGroup(['type' => 'group_type_1']);
+    for ($i = 1;$i < 4;$i++) {
+      $media = $this->createMediaEntity('document', [
+        'id' => 'media_' . $i,
+        'name' => 'Media ' . $i,
+      ]);
+      $group->addRelationship($media, 'group_media:document');
+    }
+    $user = $this->createUser([
+      'view the administration theme',
+      'access administration pages',
+      'access group overview',
+    ]);
+    $group->addMember($user);
+
+    // Go to the page and check the links added to the view.
+    $this->drupalLogin($user);
+    $this->drupalGet("group/{$group->id()}/media");
+    $assert_session = $this->assertSession();
+
+    // Check table header.
+    $table = $assert_session->elementExists('css', 'table');
+    $table_header = $assert_session->elementExists('css', 'thead', $table);
+    $rows = $table_header->findAll('css', 'tr');
+    $cols = $rows[0]->findAll('css', 'th');
+    $this->assertEquals('Media name', $cols[0]->getText());
+    $this->assertEquals('Bundle', $cols[1]->getText());
+    $this->assertEquals('Status', $cols[2]->getText());
+    $this->assertEquals('Publisher', $cols[3]->getText());
+    // Support for different vesrions of groupmedia.
+    $this->assertEquals($this->isGroupRelationShipType() ? 'Operations' : 'Dropbutton', $cols[4]->getText());
+
+    // Check that rows contain new links for operations in Collabora.
+    $table_body = $assert_session->elementExists('css', 'tbody', $table);
+    $rows = $table_body->findAll('css', 'tr');
+    $i = 0;
+    foreach(Media::loadMultiple() as $media) {
+      $cols = $rows[$i]->findAll('css', 'td');
+      $this->assertEquals($media->getName(), $cols[0]->getText());
+      $this->assertEquals($media_type->label(), $cols[1]->getText());
+      $this->assertEquals('Yes', $cols[2]->getText());
+      $this->assertEquals('Anonymous', $cols[3]->getText());
+      $operation_links = $cols[4]->findAll('css', 'a');
+      $this->assertEquals('Preview in Collabora', $operation_links[0]->getText());
+      $this->assertEquals(
+        Url::fromRoute(
+          'collabora-online.view',
+          [
+            'media' => $media->id()
+          ],
+          [
+            'query' => [
+              'destination' => "/group/{$group->id()}/media",
+            ]
+          ]
+        )->toString(),
```

Comment:
Do we need to build the url like this, or can we hard-code it as a string?
Are we worried that the url might change in the future?
Maybe it will? `cool/view/*` looks a bit odd.
It could be more relevant if the collabora_online_group will live in a different package in the future, and needs to work with different versions of the main module.

Reply:
For the destination: I think I understand why this is added, but does this actually work?
Normally this parameter tells where to return on a form submit.
But here we don't submit anything.

Reply:
About building urls this way the tests won't fail if some configuration about pipeline changes.
For the destination views add this, and we don't need to test the url here, the test is about links added to a view, so we just need to check that links are as we expect.

Reply:
Not sure about pipeline changes that would change the outcome, if we use the docker image, and also if the path is not absolute. But let's keep it as is, I don't have a strong enough conviction against it, and it works.